### ISSUE TITLE
Incorrect algebra to string with negation in filter

### ISSUE THREAD

**Issue by rubensworks** [joachimvh/SPARQLAlgebra.js#30]
When parsing and re-serializing the following query:

```sparql
PREFIX owl: <http://www.w3.org/2002/07/owl#>
select * where { { ?c a owl:Class . } FILTER ( ! (BOUND(?c) && ( REGEX(str(?c), "^toto","i")))) } limit 10
```

the filter is transformed to

```sparql
FILTER( ! (BOUND(?c)) && ( REGEX(STR(?c), "^toto", "i")))
```

I've checked the SPARQL parser, and it correctly outputs the following:

```json
{
  "queryType": "SELECT",
  "variables": [
    "*"
  ],
  "where": [
    {
      "type": "group",
      "patterns": [
        {
          "type": "bgp",
          "triples": [
            {
              "subject": "?c",
              "predicate": "http://www.w3.org/1999/02/22-rdf-syntax-ns#type",
              "object": "http://www.w3.org/2002/07/owl#Class"
            }
          ]
        }
      ]
    },
    {
      "type": "filter",
      "expression": {
        "type": "operation",
        "operator": "!",
        "args": [
          {
            "type": "operation",
            "operator": "&&",
            "args": [
              {
                "type": "operation",
                "operator": "bound",
                "args": [
                  "?c"
                ]
              },
              {
                "type": "operation",
                "operator": "regex",
                "args": [
                  {
                    "type": "operation",
                    "operator": "str",
                    "args": [
                      "?c"
                    ]
                  },
                  "\"^toto\"",
                  "\"i\""
                ]
              }
            ]
          }
        ]
      }
    }
  ],
  "limit": 10,
  "type": "query",
  "prefixes": {
    "owl": "http://www.w3.org/2002/07/owl#"
  }
}
```

So I assume that this library incorrect handles serialization of scoped negations.

This causes the problem reported here: https://github.com/comunica/comunica/issues/304

**Comment by joachimvh** [comment]
This actually seems to be a problem with the SPARQL.js library. Will leave this issue open until I fixed it there.

**Comment by joachimvh** [comment]
Fixed in b9c37a6b55aad0c94f716ed9477699d948c9efa2
